synapse/handlers/cas_handler.py

```diff
@@ -127,31 +154,46 @@ def _parse_cas_response(
             )
         return user, attributes
 
-    async def _on_successful_auth(
-        self,
-        username: str,
-        request: SynapseRequest,
-        client_redirect_url: str,
-        user_display_name: Optional[str] = None,
+    def get_redirect_url(self, service_redirect_endpoint: str, **kwargs) -> str:
+        """
+        Generates a URL to the CAS server where the client should be redirected.
+
+        Args:
+            service_redirect_endpoint: The homeserver endpoint to redirect
+                the client to after successful SSO negotiation.
```

Comment:
some words on what this might be would be helpful to follow the flow. "This might be `/_matrix/client/r0/login/cas/ticket` for the login flow or `/_matrix/client/r0/auth/cas/ticket` for the UIA flow." or something?

Reply:
This comment is overcome by refactoring.